src/store/wallet/wallet-selectors.ts

```diff
@@ -1,29 +1,45 @@
 import { useEffect, useMemo, useState } from 'react';
 import { useSelector } from 'react-redux';
 
-import { emptyWalletAccount } from '../../interfaces/wallet-account.interface';
+import { emptyWalletAccountState } from '../../interfaces/wallet-account-state.interface';
+import { WalletAccountInterface } from '../../interfaces/wallet-account.interface';
 import { TEZ_TOKEN_METADATA } from '../../token/data/tokens-metadata';
 import { emptyTokenMetadata } from '../../token/interfaces/token-metadata.interface';
 import { emptyToken, TokenInterface } from '../../token/interfaces/token.interface';
 import { WalletRootState, WalletState } from './wallet-state';
 
 export const useHdAccountsListSelector = () =>
-  useSelector<WalletRootState, WalletState['hdAccounts']>(({ wallet }) => wallet.hdAccounts);
+  useSelector<WalletRootState, WalletAccountInterface[]>(({ wallet }) =>
+    wallet.hdAccounts.map(
+      ({ activityGroups: _activityGroups, pendingActivities: _pendingActivities, ...restProps }) => restProps
+    )
+  );
 
 export const useIsAuthorisedSelector = () => useHdAccountsListSelector().length > 0;
 
-export const useSelectedAccountSelector = () => {
+const useSelectedAccountStateSelector = () => {
   const { hdAccounts, selectedAccountPublicKeyHash } = useSelector<WalletRootState, WalletState>(
     ({ wallet }) => wallet
   );
 
   // TODO: OPTIMIZE SELECTED ACCOUNT SELECTOR ASAP
   return useMemo(
-    () => hdAccounts.find(({ publicKeyHash }) => publicKeyHash === selectedAccountPublicKeyHash) ?? emptyWalletAccount,
+    () =>
+      hdAccounts.find(({ publicKeyHash }) => publicKeyHash === selectedAccountPublicKeyHash) ?? emptyWalletAccountState,
     [hdAccounts, selectedAccountPublicKeyHash]
   );
 };
 
+export const useSelectedAccountSelector = () => {
+  const { name, publicKey, publicKeyHash, tezosBalance, tokensList } = useSelectedAccountStateSelector();
+
+  return { name, publicKey, publicKeyHash, tezosBalance, tokensList };
+};
+
+export const useSelectedAccountPendingActivities = () => useSelectedAccountStateSelector().pendingActivities ?? [];
+
+export const useSelectedAccountActivityGroups = () => useSelectedAccountStateSelector().activityGroups?.data ?? [];
```

Comment:
`pendingActivities`, `activityGroups` and `activityGroups.data` could not be not defined.  
Remove `?.` and `??`

Reply:
Sorry, they can if you start with storage from the previous version
Then crashing with error like `state.activityGroups is undefined` occurs

Reply:
This's a common problem and does not relate to this specific case.  
Let's use correct types and fix this problem when "persisting" state from local storage (it will be extended with initial state)